File: backend/src/main/scala/microservices/content/objects/ExploreSuggestionListPlannerResponse.scala
package com.typesafe.travel.content.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class ExploreSuggestionListPlannerResponse(suggestions: List[ExploreSuggestionPlannerResponse])
object ExploreSuggestionListPlannerResponse:
  given sourceEncoder: Encoder[ExploreSuggestionListPlannerResponse] = deriveEncoder
  given sourceDecoder: Decoder[ExploreSuggestionListPlannerResponse] = deriveDecoder

File: frontend/src/microservices/content/objects/ExploreSuggestionListPlannerResponse.ts
// ExploreSuggestionListPlannerResponse：content 域探索推荐列表返回对象。

import type { ExploreSuggestionPlannerResponse } from './ExploreSuggestionPlannerResponse'
export type ExploreSuggestionListPlannerResponse = {
  suggestions: ExploreSuggestionPlannerResponse[]
}
export const exploreSuggestionListPlannerResponseFromJson = (json: string): ExploreSuggestionListPlannerResponse =>
  JSON.parse(json) as ExploreSuggestionListPlannerResponse
export const exploreSuggestionListPlannerResponseToJson = (value: ExploreSuggestionListPlannerResponse): string =>
  JSON.stringify(value)
